### PR TITLE
Introduce Thunder

### PR DESCRIPTION
**Pull request by D4nte**

### CODE REVIEW

**Comment by D4nte** [comment]
I went for `thor` and `thunder` but we could very well change it to `lib` `cli` or `thor-lib` `thor-cli`....

File: thor/src/transaction.rs [hunk 1]
```diff
@@ -38,11 +38,20 @@ impl FundOutput {
     }
 }
 
+#[cfg_attr(feature = "serde", derive(serde::Serialize, serde::Deserialize))]
 #[derive(Clone, PartialEq, Debug)]
 pub struct FundingTransaction {
-    inner: Transaction,
+    pub inner: Transaction,
```
Comment (luckysori) [inline thor/src/transaction.rs:44]: :wrench: Better if you define a `txid()` method and we don't reveal the internal structure of this struct.

**Review by luckysori — APPROVED** [review]
One non-blocking request, but it looks good to me.

Note that I think your change to the `SplitOutputs` may be obsoleted if we merge #37 as is. In that PR I remove `SplitOutputs` altogether.

**Comment by D4nte** [comment]
Merging now as it is hard to rebase. Taking action on the comment above in new PR.

File: thor/src/transaction.rs (hunk 1 above)
Comment (D4nte) [reply thor/src/transaction.rs:44]: Yes, good thinking

**Pull request merged**